### PR TITLE
real world testing report 2023

### PR DESCRIPTION
**Pull request by bradymiller**
real world testing report 2023

Still working on this:
3 metrics done. 2 more left to do.

### CODE REVIEW

File: interface/reports/rwt_2023_report.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+}
+
+// dates for this report are hard-coded (see header for details)
+$begin_date = '2023-04-01';
+$end_date = '2023-09-30';
+
+// can override dates below for development/testing purposes
+$begin_date = '2022-10-01';
+$end_date = '2022-12-01';
```
Comment (bradymiller) [inline interface/reports/rwt_2023_report.php:39]: Not I am overriding the dates here for now just for development/testing purposes. When this goes in the codebase, will instead use the official hard-coded dates of 2023-04-01 to 2023-09-30

File: src/Reports/RealWorldTesting.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    public function renderReport(): string
+    {
+        $output = text(xl('Date') . ': ' . oeFormatShortDate()) . '<br /><br />';
+        $output .= "<span class='font-weight-bold'>" . xlt('Metric 1') . '</span><br />';
+        $output .= text($this->metric1()) . '<br /><br />';
+        $output .= "<span class='font-weight-bold'>" . xlt('Metric 2') . '</span><br />';
+        $output .= nl2br(text(implode("\n", $this->metric2()))) . '<br /><br />';
+        $output .= "<span class='font-weight-bold'>" . xlt('Metric 3') . '</span><br />';
+        $output .= $this->metric3() . '<br /><br />';
```
Comment (bradymiller) [inline src/Reports/RealWorldTesting.php:34]: text()

File: src/Reports/RealWorldTesting.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        $output .= "<span class='font-weight-bold'>" . xlt('Metric 4') . '</span><br />';
+        $output .= $this->metric4() . '<br /><br />';
```
Comment (bradymiller) [inline src/Reports/RealWorldTesting.php:36]: text()

File: interface/reports/rwt_2023_report.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    </style>
+</head>
+
+<body class='body_top'>
+
+<span class='title'><?php echo xlt('2023 Real World Testing Report'); ?></span>
+
+<?php if (empty($_POST['start_button'])) { ?>
+    <div class='mt-2 alert alert-primary' role='alert'>
+        <?php echo xlt("This report is required for OpenEMR instances in the United States that utilize ONC 2015 certification. This reports collects metrics that are used in Real World Testing that are required for the OpenEMR Foundation to maintain the ONC 2015 certification. This report calculates metrics from April 1, 2023 to September 30, 2023. Please run this report sometime in October or November of 2023 and then print it to a pdf and email the pdf to the OpenEMR Foundation at hello@open-emr.org. In the email, please confirm your practice is in the United States and state the clinical setting of your practice (this can be 'Primary/Specialty Care' setting, 'Behavioral Health Care' setting, or any other setting).") ?>
```
Comment (stephenwaite) [inline interface/reports/rwt_2023_report.php:62]: bradys-panda

**Comment by bradymiller** [comment]
testing well and bringing in :)

**Comment by sjpadgett** [comment]
Thank you for doing this @bradymiller 
Considerable amount of work which we appreciate.

**Pull request merged**